File: dto-models/src/main/java/com/octopus/dtomodels/UserDTO.java
package com.octopus.dtomodels;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import java.io.Serializable;
import java.util.Date;

@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserDTO implements Serializable {
    private String id;
    @Email
    @NotBlank
    private String email;
    private String firstName;
    private String lastName;
    @JsonProperty(value = "password", access = JsonProperty.Access.WRITE_ONLY)
    private String password;
    private String phoneNumber;
    private Date birthday;
    private Boolean gender;
    private Boolean active;
    private Date lastActive;
    private String avatar;
    private String refreshToken;
    private Boolean enabled;
    private Date createdDate;
    private Date updatedDate;
}

File: message-service/src/dtos/user.dto.ts
export type UserDTO = {
  id: string;
  firstName?: string;
  lastName?: string;
  email: string;
  phoneNumber?: string;
  birthday?: string;
  gender?: boolean;
  active?: boolean;
  lastActive?: string;
  avatar?: string;
  enabled?: boolean;
  createdDate?: string;
  updatedDate?: string;
};

export type DeviceDTO = {
  deviceID: string;
  pushProvider?: string;
};

export type OwnUserDTO = UserDTO & {
  connections?: string[];
  devices: DeviceDTO[];
};
